src/bridge/MatrixHandler.ts

```diff
@@ -811,12 +811,20 @@ export class MatrixHandler {
         const [targetRoom] = await this.ircBridge.getStore().getIrcChannelsForRoomId(event.room_id);
         if (command === "nick") {
             const newNick = args[0];
+            if (newNick?.length > 0) {
```

Comment:
I think this does the opposite of what it should do, rejecting all newNicks with a length greater 0.
```suggestion
            if (!(newNick?.length > 0)) {
```

Reply:
Doh!